### ISSUE TITLE
Unexpected coverage drop with 3.3.0 (and using the sln option)

### ISSUE THREAD

**Issue by riezebosch** [stryker-mutator/stryker-net#2362]
**Describe the bug**
From the [normal coverage report](https://app.codecov.io/gh/riezebosch/CloudEventify/tree/main/MassTransit/CloudEventify.MassTransit) is shows clearly that my SUT is covered, but still stryker thinks otherwise and refuses to kill some mutants.

**Logs**
[2022-12-23.11-16-54.zip](https://github.com/stryker-mutator/stryker-net/files/10294506/2022-12-23.11-16-54.zip)

**Expected behavior**

**Desktop (please complete the following information):**
 - OS: macOS
 - Type of project: core
 - Framework Version: 6.0/7.0
 - Stryker Version: 3.3.0

**Additional context**
Source project: https://github.com/riezebosch/CloudEventify

**Comment by Liam-Rougoor** [comment]
It looks like if I run Stryker on just the MassTransit project, it does mark the mutants as killed correctly. I'll investigate the solution run.
![image](https://user-images.githubusercontent.com/35850587/209358304-587424a7-2b2e-4235-912a-91279e4609ee.png)

**Comment by Liam-Rougoor** [comment]
@riezebosch Could you provide the logs for the Stryker run that does succeed? We noticed a warning where some projects 'extensions' could not be found:
`2022-12-23T11:17:06.6775970+01:00  [VRB] Runner 2: [Warning] Could not find extensions: /Users/mriezebosch/git/CloudEventify/Rebus/CloudEventify.Rebus.IntegrationTests/bin/Debug/net6.0,/Users/mriezebosch/git/CloudEventify/MassTransit/CloudEventify.MassTransit.IntegrationTests/bin/Debug/net6.0,/Users/mriezebosch/git/CloudEventify/DaprApp.IntegrationTests/bin/Debug/net6.0,/Users/mriezebosch/git/CloudEventify/Dapr/CloudEventify.Dapr.IntegrationTests/bin/Debug/net6.0 (8f742cab)`
We would like to compare logs to see if we can find the issue there.

**Comment by rouke-broersma** [comment]
@Liam-Rougoor that warning comes from vstest and is expected.

**Comment by yaelkeemink** [comment]
We are having the same issue, In some cases (not all) Stryker does not seem to interpret the code coverage correctly.
[logfile.txt](https://github.com/stryker-mutator/stryker-net/files/10487322/log-20230124.txt)

Are there any updates on this issue?

**Comment by causa-prima** [comment]
I think I might encounter the same behavior:

Run against the complete solution, details for a specific project in it:
![Stryker_full_solution_run](https://user-images.githubusercontent.com/2970704/228155289-8730bd19-0311-4dcb-be2e-754555fa9585.png)

Run for only the project from the screenshot above:
![Stryker_single_project_run](https://user-images.githubusercontent.com/2970704/228155755-0bb7b345-3308-435c-90a6-64b3ad2335cf.png)

Because of confidentiality, I can't provide detailed logs. How else could I help? Or can you already reproduce the issue?

**Comment by riezebosch** [comment]
stryker@3.7.1
```
[10:44:25 INF] 99 mutants created
[10:44:25 INF] Capture mutant coverage using 'CoverageBasedTest' mode.
[11:00:55 INF] 48    mutants got status NoCoverage.   Reason: Not covered by any test.
[11:00:55 INF] 26    mutants got status Ignored.      Reason: Removed by block already covered filter
[11:00:55 INF] 74    total mutants are skipped for the above mentioned reasons
[11:00:55 INF] 25    total mutants will be tested
```

stryker@improve_solution_handling
```
[11:18:16 INF] 99 mutants created
[11:18:16 INF] Capture mutant coverage using 'CoverageBasedTest' mode.
[12:28:02 INF] 18    mutants got status NoCoverage.   Reason: Not covered by any test.
[12:28:02 INF] 26    mutants got status Ignored.      Reason: Removed by block already covered filter
[12:28:02 INF] 44    total mutants are skipped for the above mentioned reasons
[12:28:02 INF] 55    total mutants will be tested
```

Looking good! No coverage on expected cases, like forcefully implemented overloaded interfaces and startup/program classes.

**Comment by dupdob** [comment]
thanks for the feedback

**Comment by richardwerkman** [comment]
This should have been fixed by @3.9.0
